lib/ex_doc.ex

```diff
@@ -112,9 +136,30 @@ defmodule ExDoc do
   end
 
   # Helpers
+  defp nilify_options(options) do
+    for {field, value} <- options do
+      case value do
+        "" -> {field, nil}
+        _  -> {field, value}
+      end
+    end
+  end
+
+  defp normalize_option(options, field, default) do
+    case Keyword.fetch(options, field) do
+      {:ok, value} when field in [:source_ref, :source_url_pattern] and value in [nil, false] ->
+        default
+      {:ok, value} ->
+        value
+      :error ->
+        default
+    end
+  end
 
   defp normalize_options(options) do
-    pattern = options[:source_url_pattern] || guess_url(options[:source_url], options[:source_ref] || "master")
```

Comment:
This code is doing exactly what the old code does, except the new code is much more verbose. Why change it?